### ISSUE TITLE
Role hangs in its execution at Rule `3.5.3.2.1` until manually terminated when applied to new Ubuntu VM over SSH

### ISSUE THREAD

**Issue by estenrye** [darkwizard242/cis_ubuntu_2004#5]
# SUMMARY

When running this role from a remote jumpbox, Ansible connection is terminated because Rule `3.5.3.2.1` is applied before applying Rule `3.5.3.2.4`.  

## Rule 3.5.3.2.1

Rule `3.5.3.2.1` configures the default policies for the `INPUT`, `FORWARD` and `OUTPUT` chains to `DROP`.  Because it is applied before Rule `3.5.3.2.4` which applies rules to allow SSH communication it effectively terminates the SSH session being used to apply the role.

https://github.com/darkwizard242/cis_ubuntu_2004/blob/aaccda1ce3c10e599a16d2d751741d0481441496/tasks/section_03.yml#L860-L888

## Rule 3.5.3.2.4

This rule creates iptables rules that allow inbound SSH and outbound DNS requests.  I believe moving its execution before Rule `3.5.3.2.1` would resolve the issue.

https://github.com/darkwizard242/cis_ubuntu_2004/blob/aaccda1ce3c10e599a16d2d751741d0481441496/tasks/section_03.yml#L949-L968

 # ISSUE TYPE

- Bug Report

# ANSIBLE VERSION

```
ansible 2.10.2
  config file = None
  configured module search path = ['/home/esten/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/esten/.local/lib/python3.8/site-packages/ansible
  executable location = /home/esten/.local/bin/ansible
  python version = 3.8.5 (default, Jul 28 2020, 12:59:40) [GCC 9.3.0]
```

 # STEPS TO REPRODUCE

1. Provision a brand new Ubuntu 20.04 Minimal Server Virtual Machine
2. Apply this role to the new Virtual Machine over SSH.

 # EXPECTED RESULTS

Role should successfully complete execution

 # ACTUAL RESULTS

Role hangs in its execution at Rule `3.5.3.2.1` until manually terminated.

# WORKAROUND

```yaml
- hosts: targets
  roles:
    - darkwizard242.cis_ubuntu_2004
  vars:
    ubuntu_2004_cis_time_synchronization: chrony
    ubuntu_2004_cis_firewall: iptables
    ubuntu_2004_cis_section3_rule_3_5_3_2_1: false
  become: true

- hosts: targets
  roles:
    - darkwizard242.cis_ubuntu_2004
  vars:
    preliminary_appliance: false
    ubuntu_2004_cis_section1: false
    ubuntu_2004_cis_section2: false
    ubuntu_2004_cis_section3: true
    ubuntu_2004_cis_section4: false
    ubuntu_2004_cis_section5: false
    ubuntu_2004_cis_section6: false
    ubuntu_2004_cis_firewall: iptables
  become: true
```

**Comment by darkwizard242** [comment]
@estenrye - Thanks for reporting.

Have you tried utilizing the vars set to `false` as shown (for example) in the [playbook here](https://github.com/darkwizard242/cis_ubuntu_2004/blob/master/playbook-examples/playbook_with_custom_firewall_changes.yml).

This should allow for **Rule 3.5.3.2.1** to not be applied as the `when` conditions would be false.

The reason I am hesitant about moving them is because I have been trying to keep the tasks and their execution aligned with how they are structured in the CIS benchmarks as most folks go from one control to the next.

**Comment by estenrye** [comment]
yeah, that's what I ended up doing.

I'd agree with that approach if this role were focused solely in an audit capacity, but this role appears to have more of a remediation focus than audit.  Because of its focus on remediation, reducing the amount of configuration required to apply sane defaults makes sense to me even though it causes a bit of cognitive burden by being out of order.

I also noticed that you don't persist the firewall rules.  Because they are not persisted, the remediation for those rules does not survive a reboot.  Did I miss a variable that persists these settings?

**Comment by darkwizard242** [comment]
@estenrye - in the latest version, i.e. "**_CIS Ubuntu Linux 20.04 LTS Benchmark v1.1.0_"**, the mentioned controls have apparently been moved to:

- 3.5.3.2.3 Ensure iptables default deny firewall policy (Automated)
- 3.5.3.2.4 Ensure iptables firewall rules exist for all open ports (Automated)

As I am in developing for the "**_CIS Ubuntu Linux 20.04 LTS Benchmark v1.1.0_**", I can address your request in there (_by changing the order for these specific tasks_) - but between day job, family and other commitments; it's been somewhat slow. Though, I do plan to complete the role's development to comply with CIS 1.1.0 version within a month or so.

Ah, I wasn't aware that the rules weren't persisting post-reboot. I will do some testing over the weekend. Thanks for bringing that up.

**Comment by darkwizard242** [comment]
@estenrye - Tasks have been moved in the requested order within Release [2.0.0](https://github.com/darkwizard242/cis_ubuntu_2004/releases/tag/2.0.0)

**Comment by estenrye** [comment]
@darkwizard242 cool, I'll try and get something on the board to test version 2.0.0 soon.

**Comment by estenrye** [comment]
Resolved with #8
